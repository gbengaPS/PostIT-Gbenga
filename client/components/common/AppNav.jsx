import React from 'react';
import { Link } from 'react-router-dom';
import UserGroups from '../group/UserGroups';

/**
 * @description AppNav component. It contains navigation links
 * to app functionalities
 *
 * @param { object } props -prop object
 *
 * @returns { object } react element
 */
const AppNav = () => (
  (
    <div>
    <div
      className="col m2 component-container hide-on-med-and-down user-side-nav">
      <ul className="text-center collapsible" data-collapsible="accordion">
      <li className="big">
        <div className="collapsible-header">
        <Link to="/dashboard">
        <i className="material-icons">home</i>
           &nbsp;&nbsp;Dashboard
        </Link>
        </div>
        </li>

        <li className="big collapsible-header">
          <Link to="/group/create" id="group-create">
          <i className="material-icons">group_add</i>
           &nbsp;&nbsp;Create group
          </Link>
        </li>

        <li className="big">
          <div className="collapsible-header">
          <Link to="#">
            <i className="material-icons">group</i>
            &nbsp;&nbsp;My Groups
          </Link>
          </div>
        <div className="collapsible-body scrollable-component">
          <UserGroups />
        </div>
        </li>
      </ul>
    </div>
    </div>
  )
);

export default AppNav;
