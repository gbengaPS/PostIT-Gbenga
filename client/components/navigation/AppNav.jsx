import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import UserGroups from '../group/UserGroups';

const AppNav = (props) => {
  let appHeader;
  if (props.useHeader) {
    appHeader = (
      <div className="dashboard-header row">
      <a href="#" className="right dropdown-button"
      data-activates="group-more">
      <i className="material-icons">more_vert</i>
      <i className="material-icons">person</i>
      <span className="group-members"> {props.numberOfGroupMembers}</span></a>
      <span className="bold">{props.groupName}</span>
    </div>
    );
  }
  return (
    <div>
    <div
      className="col m3 component-container hide-on-med-and-down user-side-nav">
      <ul className="text-center collapsible" data-collapsible="accordion">
      <li className="big">
        <div className="collapsible-header">
        <i className="material-icons">home</i>
        <Link to="/dashboard">
           &nbsp;&nbsp;Dashboard
          </Link>
        </div>
        </li>

        <li className="big collapsible-header">
        <i className="material-icons">group_add</i>
          <Link to="/group/create">
           &nbsp;&nbsp;Create group
          </Link>
        </li>

        <li className="big">
          <div className="collapsible-header">
          <i className="material-icons">group</i>
          <Link to="#">&nbsp;&nbsp;My Groups</Link>
          </div>
        <div className="collapsible-body">
          <UserGroups />
        </div>
        </li>
      </ul>
    </div>
      <div className="col s10 offset-s1 m6 component-container">
        {appHeader}
      </div>
    </div>
  );
};

AppNav.propTypes = {
  numberOfGroupMembers: PropTypes.number,
  groupName: PropTypes.string,
  useHeader: PropTypes.string,
};

export default AppNav;
