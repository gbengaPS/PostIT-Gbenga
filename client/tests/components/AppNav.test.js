import React from 'react';
import AppNav from '../../components/common/AppNav';

describe('AppNav component ', () => {
  it('should render correctly', () => {
    expect(shallow(<AppNav />)).toMatchSnapshot();
  });
});
