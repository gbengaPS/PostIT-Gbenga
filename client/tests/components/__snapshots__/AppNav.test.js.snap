// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav component  should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppNav
    useHeader={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col m2 component-container hide-on-med-and-down user-side-nav"
      >
        <ul
          className="text-center collapsible"
          data-collapsible="accordion"
        >
          <li
            className="big"
          >
            <div
              className="collapsible-header"
            >
              <Link
                replace={false}
                to="/dashboard"
              >
                <i
                  className="material-icons"
                >
                  home
                </i>
                  Dashboard
              </Link>
            </div>
          </li>
          <li
            className="big collapsible-header"
          >
            <Link
              id="group-create"
              replace={false}
              to="/group/create"
            >
              <i
                className="material-icons"
              >
                group_add
              </i>
                Create group
            </Link>
          </li>
          <li
            className="big"
          >
            <div
              className="collapsible-header"
            >
              <Link
                replace={false}
                to="#"
              >
                <i
                  className="material-icons"
                >
                  group
                </i>
                  My Groups
              </Link>
            </div>
            <div
              className="collapsible-body scrollable-component"
            >
              <Connect(UserGroups) />
            </div>
          </li>
        </ul>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="text-center collapsible"
          data-collapsible="accordion"
        >
          <li
            className="big"
          >
            <div
              className="collapsible-header"
            >
              <Link
                replace={false}
                to="/dashboard"
              >
                <i
                  className="material-icons"
                >
                  home
                </i>
                  Dashboard
              </Link>
            </div>
          </li>
          <li
            className="big collapsible-header"
          >
            <Link
              id="group-create"
              replace={false}
              to="/group/create"
            >
              <i
                className="material-icons"
              >
                group_add
              </i>
                Create group
            </Link>
          </li>
          <li
            className="big"
          >
            <div
              className="collapsible-header"
            >
              <Link
                replace={false}
                to="#"
              >
                <i
                  className="material-icons"
                >
                  group
                </i>
                  My Groups
              </Link>
            </div>
            <div
              className="collapsible-body scrollable-component"
            >
              <Connect(UserGroups) />
            </div>
          </li>
        </ul>,
        "className": "col m2 component-container hide-on-med-and-down user-side-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="/dashboard"
                >
                  <i
                    className="material-icons"
                  >
                    home
                  </i>
                    Dashboard
                </Link>
              </div>
            </li>,
            <li
              className="big collapsible-header"
            >
              <Link
                id="group-create"
                replace={false}
                to="/group/create"
              >
                <i
                  className="material-icons"
                >
                  group_add
                </i>
                  Create group
              </Link>
            </li>,
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="#"
                >
                  <i
                    className="material-icons"
                  >
                    group
                  </i>
                    My Groups
                </Link>
              </div>
              <div
                className="collapsible-body scrollable-component"
              >
                <Connect(UserGroups) />
              </div>
            </li>,
          ],
          "className": "text-center collapsible",
          "data-collapsible": "accordion",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="/dashboard"
                >
                  <i
                    className="material-icons"
                  >
                    home
                  </i>
                    Dashboard
                </Link>
              </div>,
              "className": "big",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/dashboard"
                >
                  <i
                    className="material-icons"
                  >
                    home
                  </i>
                    Dashboard
                </Link>,
                "className": "collapsible-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons"
                    >
                      home
                    </i>,
                    "  Dashboard",
                  ],
                  "replace": false,
                  "to": "/dashboard",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "home",
                      "className": "material-icons",
                    },
                    "ref": null,
                    "rendered": "home",
                    "type": "i",
                  },
                  "  Dashboard",
                ],
                "type": [Function],
              },
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                id="group-create"
                replace={false}
                to="/group/create"
              >
                <i
                  className="material-icons"
                >
                  group_add
                </i>
                  Create group
              </Link>,
              "className": "big collapsible-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons"
                  >
                    group_add
                  </i>,
                  "  Create group",
                ],
                "id": "group-create",
                "replace": false,
                "to": "/group/create",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "group_add",
                    "className": "material-icons",
                  },
                  "ref": null,
                  "rendered": "group_add",
                  "type": "i",
                },
                "  Create group",
              ],
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="collapsible-header"
                >
                  <Link
                    replace={false}
                    to="#"
                  >
                    <i
                      className="material-icons"
                    >
                      group
                    </i>
                      My Groups
                  </Link>
                </div>,
                <div
                  className="collapsible-body scrollable-component"
                >
                  <Connect(UserGroups) />
                </div>,
              ],
              "className": "big",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="#"
                  >
                    <i
                      className="material-icons"
                    >
                      group
                    </i>
                      My Groups
                  </Link>,
                  "className": "collapsible-header",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="material-icons"
                      >
                        group
                      </i>,
                      "  My Groups",
                    ],
                    "replace": false,
                    "to": "#",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "group",
                        "className": "material-icons",
                      },
                      "ref": null,
                      "rendered": "group",
                      "type": "i",
                    },
                    "  My Groups",
                  ],
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(UserGroups) />,
                  "className": "collapsible-body scrollable-component",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col m2 component-container hide-on-med-and-down user-side-nav"
        >
          <ul
            className="text-center collapsible"
            data-collapsible="accordion"
          >
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="/dashboard"
                >
                  <i
                    className="material-icons"
                  >
                    home
                  </i>
                    Dashboard
                </Link>
              </div>
            </li>
            <li
              className="big collapsible-header"
            >
              <Link
                id="group-create"
                replace={false}
                to="/group/create"
              >
                <i
                  className="material-icons"
                >
                  group_add
                </i>
                  Create group
              </Link>
            </li>
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="#"
                >
                  <i
                    className="material-icons"
                  >
                    group
                  </i>
                    My Groups
                </Link>
              </div>
              <div
                className="collapsible-body scrollable-component"
              >
                <Connect(UserGroups) />
              </div>
            </li>
          </ul>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="text-center collapsible"
            data-collapsible="accordion"
          >
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="/dashboard"
                >
                  <i
                    className="material-icons"
                  >
                    home
                  </i>
                    Dashboard
                </Link>
              </div>
            </li>
            <li
              className="big collapsible-header"
            >
              <Link
                id="group-create"
                replace={false}
                to="/group/create"
              >
                <i
                  className="material-icons"
                >
                  group_add
                </i>
                  Create group
              </Link>
            </li>
            <li
              className="big"
            >
              <div
                className="collapsible-header"
              >
                <Link
                  replace={false}
                  to="#"
                >
                  <i
                    className="material-icons"
                  >
                    group
                  </i>
                    My Groups
                </Link>
              </div>
              <div
                className="collapsible-body scrollable-component"
              >
                <Connect(UserGroups) />
              </div>
            </li>
          </ul>,
          "className": "col m2 component-container hide-on-med-and-down user-side-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="big"
              >
                <div
                  className="collapsible-header"
                >
                  <Link
                    replace={false}
                    to="/dashboard"
                  >
                    <i
                      className="material-icons"
                    >
                      home
                    </i>
                      Dashboard
                  </Link>
                </div>
              </li>,
              <li
                className="big collapsible-header"
              >
                <Link
                  id="group-create"
                  replace={false}
                  to="/group/create"
                >
                  <i
                    className="material-icons"
                  >
                    group_add
                  </i>
                    Create group
                </Link>
              </li>,
              <li
                className="big"
              >
                <div
                  className="collapsible-header"
                >
                  <Link
                    replace={false}
                    to="#"
                  >
                    <i
                      className="material-icons"
                    >
                      group
                    </i>
                      My Groups
                  </Link>
                </div>
                <div
                  className="collapsible-body scrollable-component"
                >
                  <Connect(UserGroups) />
                </div>
              </li>,
            ],
            "className": "text-center collapsible",
            "data-collapsible": "accordion",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="collapsible-header"
                >
                  <Link
                    replace={false}
                    to="/dashboard"
                  >
                    <i
                      className="material-icons"
                    >
                      home
                    </i>
                      Dashboard
                  </Link>
                </div>,
                "className": "big",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/dashboard"
                  >
                    <i
                      className="material-icons"
                    >
                      home
                    </i>
                      Dashboard
                  </Link>,
                  "className": "collapsible-header",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <i
                        className="material-icons"
                      >
                        home
                      </i>,
                      "  Dashboard",
                    ],
                    "replace": false,
                    "to": "/dashboard",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "home",
                        "className": "material-icons",
                      },
                      "ref": null,
                      "rendered": "home",
                      "type": "i",
                    },
                    "  Dashboard",
                  ],
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  id="group-create"
                  replace={false}
                  to="/group/create"
                >
                  <i
                    className="material-icons"
                  >
                    group_add
                  </i>
                    Create group
                </Link>,
                "className": "big collapsible-header",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons"
                    >
                      group_add
                    </i>,
                    "  Create group",
                  ],
                  "id": "group-create",
                  "replace": false,
                  "to": "/group/create",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "group_add",
                      "className": "material-icons",
                    },
                    "ref": null,
                    "rendered": "group_add",
                    "type": "i",
                  },
                  "  Create group",
                ],
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="collapsible-header"
                  >
                    <Link
                      replace={false}
                      to="#"
                    >
                      <i
                        className="material-icons"
                      >
                        group
                      </i>
                        My Groups
                    </Link>
                  </div>,
                  <div
                    className="collapsible-body scrollable-component"
                  >
                    <Connect(UserGroups) />
                  </div>,
                ],
                "className": "big",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="#"
                    >
                      <i
                        className="material-icons"
                      >
                        group
                      </i>
                        My Groups
                    </Link>,
                    "className": "collapsible-header",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <i
                          className="material-icons"
                        >
                          group
                        </i>,
                        "  My Groups",
                      ],
                      "replace": false,
                      "to": "#",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "group",
                          "className": "material-icons",
                        },
                        "ref": null,
                        "rendered": "group",
                        "type": "i",
                      },
                      "  My Groups",
                    ],
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(UserGroups) />,
                    "className": "collapsible-body scrollable-component",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
